jmix-superset/superset-flowui/src/main/java/io/jmix/supersetflowui/DefaultGuestTokenProvider.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright 2024 Haulmont.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.jmix.supersetflowui;
+
+import com.google.common.base.Strings;
+import io.jmix.core.annotation.Internal;
+import io.jmix.core.usersubstitution.CurrentUserSubstitution;
+import io.jmix.flowui.backgroundtask.BackgroundTask;
+import io.jmix.flowui.backgroundtask.BackgroundWorker;
+import io.jmix.flowui.backgroundtask.TaskLifeCycle;
+import io.jmix.superset.SupersetTokenManager;
+import io.jmix.superset.client.model.GuestTokenBody;
+import io.jmix.superset.client.model.GuestTokenResponse;
+import io.jmix.superset.client.SupersetClient;
+import io.jmix.supersetflowui.component.SupersetDashboard;
+import io.jmix.supersetflowui.component.dataconstraint.DatasetConstrainsProvider;
+import io.jmix.supersetflowui.component.dataconstraint.DatasetConstraint;
+import jakarta.annotation.Nullable;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.ArrayUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import static io.jmix.superset.client.model.GuestTokenBody.Resource.DASHBOARD_TYPE;
+
+/**
+ * The class fetches a guest token for the {@link SupersetDashboard} component in a non-blocking manner.
+ *
+ * @see SupersetTokenManager
+ * @see SupersetClient
+ */
+@Internal
+@Component("sprset_GuestTokenHandler")
+public class DefaultGuestTokenProvider {
```

Comment:
Shouldn't this class implement the `SupersetGuestTokenProvider` interface? In this case, inside the `SupersetDashboard` you will have only one field with the tokenProvider. By default the loader will set the default provider, if the property in xml is defined or the new provider is configured using the setter from the screen, the value of the field will be replaced.

Reply:
Fixed